### DISCUSSION TITLE
Midi thru

### DISCUSSION THREAD

**Discussion by tokejuhoklee** [bangcorrupt/hacktribe#224] [Ideas]
I remember reading that you said MIDI through would be possible at some point. It's really the only thing that I need to make the electribe the perfect machine for my setup and I'm wondering whether not it will be worked on this year? Also if there's anything I could to to expedite the process or something I'd happily look into it. 

Thanks for all the great work

**Comment by bangcorrupt** [comment]
I think this should happen sometime this year.  I plan to share some assembly code that links only into the Hacktribe patch, probably a function processing MIDI input.  This would allow filtering, remapping or passing directly to the output for soft thru.  I'm working on an FX editor at the moment , so may be a while before I get started, but most of the leg work is already done.   When people sponsor Hacktribe I try to prioritise features they have asked for, but this doesn't guarantee when I'll start working on something, or if I'll be successful.

**Comment by bangcorrupt** [comment]
Thank you for your support! :smiley: 

I'll hopefully be working on this within the next couple of months.

**Reply by tokejuhoklee** [reply]
Don't think of the support as a nudge to what you wrote earlier! I've always meant to support your work and I'm in a place economically to do what I meant to do the whole time anyways!

**Comment by bangcorrupt** [comment]
Commit [0c04b4a](https://github.com/bangcorrupt/hacktribe/commit/0c04b4af28f4de181067be1fb6ae3dce7daf0401) to the `experimental` branch adds initial support for MIDI Soft Thru. 

This is now in the `main` branch.

**Reply by bangcorrupt** [reply]
This is switched off by default, there is a hidden global setting, accessible via MIDI NRPN.

See [this comment](https://github.com/bangcorrupt/hacktribe/discussions/189#discussioncomment-11748897).
